File: app/src/main/java/com/smajilbasic/kaesler/mgeprojekt/MultiplicationActivity.java
package com.smajilbasic.kaesler.mgeprojekt;

import static com.smajilbasic.kaesler.mgeprojekt.Helper.HISTORY_FILENAME;
import static com.smajilbasic.kaesler.mgeprojekt.Helper.USER_PREFERENCES;
import static com.smajilbasic.kaesler.mgeprojekt.Helper.getThemeId;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.file.Paths;

public class MultiplicationActivity extends AppCompatActivity implements View.OnClickListener {

    TextView resultBox;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SharedPreferences sharedPref = getSharedPreferences(USER_PREFERENCES,MODE_PRIVATE);
        setTheme(getThemeId(getApplication(),sharedPref));

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_multiplication);
        findViewById(R.id.calculateButton).setOnClickListener(this);
        resultBox = findViewById(R.id.result);
    }


    @Override
    public void onClick(View view) {
        if (view.getId() == R.id.calculateButton) {

            String firstInput = ((EditText) findViewById(R.id.firstNumber)).getText().toString();
            String secondInput = ((EditText) findViewById(R.id.secondNumber)).getText().toString();

            if (!firstInput.isEmpty()) {
                if (!secondInput.isEmpty()) {
                    StringBuilder builder = new StringBuilder();
                    Double firstNumber = Double.valueOf(firstInput);
                    Double secondNumber = Double.valueOf(secondInput);
                    Double result = firstNumber * secondNumber;
                    builder.append(firstInput).append(" * ").append(secondInput).append(" = ").append(result).append("\n");
                    resultBox.append(builder.toString());
                    Helper.writeHistoryEntryToFile(this, new FileEntry(firstNumber.toString(),secondNumber.toString(),result.toString(),'*'));
                } else {
                    Toast.makeText(this, getString(R.string.second_number_error_message_de), Toast.LENGTH_LONG).show();
                }
            } else {
                Toast.makeText(this, getString(R.string.first_number_error_message_de), Toast.LENGTH_LONG).show();
            }
        }
    }
}

File: app/src/main/java/com/smajilbasic/kaesler/mgeprojekt/MultiplicationActivity.kt
package com.smajilbasic.kaesler.mgeprojekt

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MultiplicationActivity : AppCompatActivity(), View.OnClickListener {
    var resultBox: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        val sharedPref = getSharedPreferences(Helper.USER_PREFERENCES, MODE_PRIVATE)
        setTheme(Helper.getThemeId(application, sharedPref))
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_multiplication)
        findViewById<View>(R.id.calculateButton).setOnClickListener(this)
        resultBox = findViewById(R.id.result)
        val actionBar = supportActionBar
        actionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onClick(view: View) {
        if (view.id == R.id.calculateButton) {
            val firstInput = (findViewById<View>(R.id.firstNumber) as EditText).text.toString()
            val secondInput = (findViewById<View>(R.id.secondNumber) as EditText).text.toString()
            if (!firstInput.isEmpty()) {
                if (!secondInput.isEmpty()) {
                    val builder = StringBuilder()
                    val firstNumber = java.lang.Double.valueOf(firstInput)
                    val secondNumber = java.lang.Double.valueOf(secondInput)
                    val result = firstNumber * secondNumber
                    builder.append(firstInput).append(" * ").append(secondInput).append(" = ").append(result).append("\n")
                    resultBox!!.append(builder.toString())
                    Helper.writeHistoryEntryToFile(this, FileEntry(firstNumber.toString(), secondNumber.toString(), result.toString(), '*'))
                } else {
                    Toast.makeText(this, getString(R.string.second_number_error_message_de), Toast.LENGTH_LONG).show()
                }
            } else {
                Toast.makeText(this, getString(R.string.first_number_error_message_de), Toast.LENGTH_LONG).show()
            }
        }
    }
}